qiskit/tools/visualization/_circuit_visualization.py

```diff
@@ -285,11 +292,15 @@ def _text_circuit_drawer(circuit, filename=None, line_length=None, reversebits=F
                    at all, set line_length=-1.
         reversebits (bool): Rearrange the bits in reverse order.
         plotbarriers (bool): Draws the barriers when they are there.
+        justify (str) : `left`, `right` or `none`. Defaults to `left`. Says how
+                        the circuit should be justified.
     Returns:
         TextDrawing: An instances that, when printed, draws the circuit in ascii art.
     """
     qregs, cregs, ops = _utils._get_instructions(circuit, reversebits=reversebits)
-    text_drawing = _text.TextDrawing(qregs, cregs, ops)
+    text_drawing = _text.TextDrawing(qregs, cregs, ops, circuit,
```

Comment:
the structure of the operations to draw is controlled by `_get_instructions` (the line before). Probably makes sense to have the `justify` parameter there.

Reply:
Thanks, this cleared things up a bit! I have moved the `justify` parameter and code to a new method `_get_layered_instructions()` in `_utils`. I made it a separate method as it is necessary to have the instructions sorted into justified layers, and the other drawers don't support this yet. We will be able to get rid of it once all the visualizers can handle justified layers.

Reply:
Hi @1ucian0 are you able to take a look at these changes so that we can get this merged?